### ISSUE TITLE
Flow  8.1.12 was accidentally tagged from the 9.0 branch

### ISSUE THREAD

**Issue by kdambekalns** [neos/flow-development-collection#3355]
Flow 8.1.12 was tagged on the 9.0 branch. :see_no_evil:

I edited the text at https://github.com/neos/flow-development-collection/releases/tag/8.1.12

Being the latest 8.1, what should we do? Drop it, or release 8.1.13 from the right branch? 

The paramaters of the release job were correct (see https://jenkins.neos.io/job/flow-release/444/parameters/), but the job failed silently (see https://jenkins.neos.io/job/flow-release/444/console).

**Comment by kitsunet** [comment]
> Drop it, or release 8.1.13 from the right branch?

Both I'd say

**Comment by kdambekalns** [comment]
Doh.

- 8.1.11, same issue? https://jenkins.neos.io/job/flow-release/437/console
- 8.1.10, too? https://jenkins.neos.io/job/flow-release/432/console
- 8.1.9? https://jenkins.neos.io/job/flow-release/424/console

Could it be those fail, because the release helpers are not in the 8.1 branch? Or something like that?

**Comment by kdambekalns** [comment]
No, actually only 8.1.12 is on the 9.0 branch, phew! The `"target_commitish": "9.0"` at the end is of no relevance (it seems), the error about Behat caused this.

**Comment by kitsunet** [comment]
Something IS very weird here though:

Note this is about the flow-development-distribution as we checkout that  one which has the build scripts:

```
18:32:08 Previous HEAD position was 440538e TASK: 7.3 upmerges
18:32:08 Switched to a new branch '8.1'
18:32:08 Branch '8.1' set up to track remote branch '8.1' from 'origin'.
18:32:08 + git reset --hard origin/8.1
18:32:08 HEAD is now at 9f27d3a TASK: 8.0 upmerges
```

But looking at this mentioned commit on GH suggests it only exists in the 8.3 branch 🤔 how can that be, unless 8.1 and 8.2 are broken?

Or GH does weird stuff, because this seems like it's there:
<img width="745" alt="Screenshot 2024-05-23 at 21 04 29" src="https://github.com/neos/flow-development-collection/assets/324408/f3bb8d4e-dd55-47c9-bf56-0d18ff841899">

**Comment by kitsunet** [comment]
Oh wait, I guess the problem is "simple" in that we composer install failed and we didn't checkout the build essentials package?

> 8:32:18   - Installing neos/buildessentials (8.1.x-dev 215c7e5): Cloning 215c7e5100b2e2122b61684f1d88a2c12ce2f561 from cache
> 18:32:18     Install of neos/buildessentials failed

**Comment by kdambekalns** [comment]
Those scripts should definitely exit with an error code when something goes wrong, and Jenkins should then fail the build!

**Comment by kitsunet** [comment]
aaand so this will run 

https://github.com/neos/flow-development-distribution/blob/8.1/Build/release.sh 

but fail for the whole tagging process beecause the essentials are not there, and tehn in the end call GH API to create a release which I guess GH then just creates on the default branch if the tag doesn't exist.

**Comment by kitsunet** [comment]
yep, I guess that solves the riddle. needs more failure checks, all else seems good.
